docs/plugins/verifier/cosign.md

```diff
@@ -1,23 +1,208 @@
+## Table of Contents
 
-# Signing
-Please refer cosign documentation on how to sign an image using cosign using [key-pair based signatures](https://docs.sigstore.dev/key_management/signing_with_self-managed_keys/) and [keyless signatures](https://docs.sigstore.dev/signing/quickstart/#keyless-signing-of-a-container).
+- [Cosign](#cosign)
+  - [Table of Contents](#table-of-contents)
+  - [Signing](#signing)
+  - [Some Caveats](#some-caveats)
+  - [Trust Policy](#trust-policy)
+    - [Scopes](#scopes)
+    - [Keys](#keys)
+      - [Supported Key Types](#supported-key-types)
+    - [Limitations](#limitations)
+  - [Demo: Multi-key, Multi-image Verification](#demo-multi-key-multi-image-verification)
+    - [Recording](#recording)
+    - [Walkthrough](#walkthrough)
+  - [Keyless Verification](#keyless-verification)
+    - [Configuration](#configuration)
+      - [Kubernetes](#kubernetes)
+      - [CLI](#cli)
+    - [Usage](#usage)
+  - [Configuration](#configuration-1)
+    - [Kubernetes](#kubernetes-1)
+    - [CLI](#cli-1)
+  - [Legacy: Key-pair based verification](#legacy-key-pair-based-verification)
+    - [Configuration](#configuration-2)
+      - [Kubernetes](#kubernetes-2)
+      - [CLI](#cli-2)
+    - [Usage](#usage-1)
+  
+## Signing
 
-# Verification
+Please refer to cosign documentation on how to sign an image using cosign using [key-pair based signatures](https://docs.sigstore.dev/key_management/signing_with_self-managed_keys/) and [keyless signatures](https://docs.sigstore.dev/signing/quickstart/#keyless-signing-of-a-container).
 
-## Key-pair based verification
-This section outlines how to use `ratify` to verify the signatures signed using key pairs.
+## Some Caveats
 
-Following is an example `ratify` config with cosign verifier. Please note the `key` refers to the public key generated by `cosign generate-key-pair` command. It is used to verify the signature signed by cosign.
+A configuration flag `cosignEnabled` is introduced to the ORAS Store configuration. This flag is required to be set to `true` in order to fetch cosign signatures stored in an OCI Image tagged with Cosign's unique convention (`<hash-algorithm>-<hash-of-subject-image>.sig`). Regardless of the flag being set, Ratify will fetch OCI 1.1-compliant cosign signatures returned via the referrers API.
+
+## Trust Policy
+
+A trust policy binds a set of verification configurations against a set of registry-reference scopes. In particular, a trust policy allows a user to define the trusted keys to use for a given set of scopes.
+
+**Sample trust policies:**
```

Comment:
should trust policy be organized by versions though there is just one version for now.

Reply:
Yes I agree that would be ideal. But I'm thinking since it's just one version maybe we introduce later if needed